roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -20,147 +20,168 @@
-  void setVerbose(bool flag=true) ;
-  void setProfile(bool flag=true) { _profile = flag ; }
-  bool setLogFile(const char* logf=nullptr) ;
-
-  /// Enable or disable the logging of function evaluations to a RooDataSet.
-  /// \see RooMinimizer::getLogDataSet().
-  /// param[in] flag Boolean flag to disable or enable the functionality.
-  void setLoggingToDataSet(bool flag=true) { _loggingToDataSet = flag ; }
-
-  /// If logging of function evaluations to a RooDataSet is enabled, returns a
-  /// pointer to a dataset with one row per evaluation of the RooAbsReal passed
-  /// to the minimizer. As columns, there are all floating parameters and the
-  /// values they had for that evaluation.
-  /// \see RooMinimizer::setLoggingToDataSet(bool).
-  RooDataSet * getLogDataSet() const { return _logDataSet.get(); }
-
-  int getPrintLevel() const { return _printLevel; }
-
-  void setMinimizerType(std::string const& type) ;
-  std::string const& minimizerType() const { return _minimizerType; }
-
-  static void cleanup() ;
-  static RooFitResult* lastMinuitFit() ;
-  static RooFitResult* lastMinuitFit(const RooArgList& varList) ;
-
-  void saveStatus(const char* label, int status) { _statusHistory.push_back(std::pair<std::string,int>(label,status)) ; }
-
-  int evalCounter() const ;
-  void zeroEvalCount() ;
-
-  ROOT::Fit::Fitter* fitter() ;
-  const ROOT::Fit::Fitter* fitter() const ;
-
-  ROOT::Math::IMultiGenFunction* getMultiGenFcn() const;
-
-  int getNPar() const ;
-
-  void applyCovarianceMatrix(TMatrixDSym const& V) ;
+   enum class FcnMode { classic, gradient, generic_wrapper };
+
+   /// Config argument to RooMinimizer ctor
+   struct Config {
+      double recoverFromNaN = 10.; // RooAbsMinimizerFcn config
+      int optConst = 0;            // RooAbsMinimizerFcn config
+      int printEvalErrors = 10;    // RooAbsMinimizerFcn config
+      int doEEWall = 1;            // RooAbsMinimizerFcn config
+      int offsetting = -1;         // RooAbsMinimizerFcn config
+      const char *logf = nullptr;  // RooAbsMinimizerFcn config
+      int nWorkers =
+         RooFit::MultiProcess::Config::getDefaultNWorkers(); // RooAbsMinimizerFcn config that can only be set in ctor
+      int parallel_gradient = 0;          // // RooAbsMinimizerFcn config that can only be set in ctor
+      int parallel_likelihood = 0;        // // RooAbsMinimizerFcn config that can only be set in ctor
+      int verbose = 0;                    // local config
+      bool profile;                       // local config
+      std::string minimizerType = "";     // local config
+      FcnMode fcnMode = FcnMode::classic; // local config
+   };
+
+   explicit RooMinimizer(RooAbsReal &function, FcnMode fcnMode = FcnMode::classic);
+   explicit RooMinimizer(RooAbsReal &function, Config &cfg);
+   explicit RooMinimizer(std::shared_ptr<RooFit::TestStatistics::RooAbsL> likelihood, Config &cfg);
+
+   ~RooMinimizer() override;
+   RooMinimizer(const RooMinimizer &min);
```

Comment:
```suggestion
   RooMinimizer(const RooMinimizer &min);
```
This is declared but never defined in the `cxx`. Can you delete it please?